lib/sasl/sasl.js

```diff
@@ -60,41 +54,50 @@ Sasl.prototype.headerReceived = function(header) {
 };
 
 Sasl.prototype._processFrame = function(frame) {
+  var self = this;
   if (frame instanceof frames.SaslMechanismsFrame) {
-    var buf = new Builder();
     var mechanism = this.mechanism;
     var mechanisms = Array.isArray(frame.saslServerMechanisms) ?
       frame.saslServerMechanisms.map(function(m) { return m.value; }) : frame.saslServerMechanisms;
     if (!u.includes(mechanisms, mechanism)) {
       this.callback(new errors.AuthenticationError(
         'SASL ' + mechanism + ' not supported by remote.'));
     }
-    if (mechanism === Sasl.Mechanism.PLAIN) {
-      debug('Sending ' + this.credentials.user + ':' + this.credentials.pass);
-      buf.appendUInt8(0); // <null>
-      buf.appendString(this.credentials.user);
-      buf.appendUInt8(0); // <null>
-      buf.appendString(this.credentials.pass);
-    } else if (mechanism === Sasl.Mechanism.ANONYMOUS) {
-      if (this.credentials.user && this.credentials.pass) {
-        console.warn(
-            'Sasl ANONYMOUS requested, but credentials provided in endpoint URI');
-      }
-      buf.appendUInt8(0); // <null>
-    } else {
-      this.callback(new errors.NotImplementedError(
-          'Only SASL PLAIN and ANONYMOUS are supported.'));
+
+    if (mechanism === Sasl.Mechanism.ANONYMOUS && this.credentials.user && this.credentials.pass) {
+      console.warn(
+          'Sasl ANONYMOUS requested, but credentials provided in endpoint URI');
```

Comment:
does it make sense for this to be in the sasl anonymous handler instead so that the knowledge of that mechanism is self-contained?

Reply:
I've gone back and forth on this. Ideally, yes, but since the credentials object is not a thing that is part of the Sasl-Anonymous mechanism it would also be weird for it to be there IMHO. Plus, it feels safer to not move that for now since it was already a part of the existing logic (making sure we're not breaking too many things at once).

Reply:
i think that's reasonable